### PR TITLE
Fix crash with tf.transpose when a is complex and conjugate is True

### LINKED ISSUE #46891: tf.transpose crashes(abort) if `a` is complex
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

**Describe the current behavior**
`tf.transpose` crashes(abort) if `a` is complex and `conjugate`=True

**Describe the expected behavior**
expect no crash

**Standalone code to reproduce the issue**
~~~python
import tensorflow as tf
tf.transpose(conjugate=True, a=complex(1))
~~~

Output:
~~~python
2021-02-03 17:58:05.565680: F ./tensorflow/core/kernels/transpose_functor.h:169] Check failed: in.dims() >= 2 (0 vs. 2)
Aborted (core dumped)
~~~

### PR DESCRIPTION
**Pull request by yongtang**
This PR tries to address the issue raised in #46891 where
tf.transpose will crash when a is complex and conjugate is True.
The issue comes from:
https://github.com/tensorflow/tensorflow/blob/57bbc5e0d4b93483b8ae853352173516f1c08018/tensorflow/core/kernels/transpose_functor.h#L169

However, as ndims < 2 has already been handled properly:
https://github.com/tensorflow/tensorflow/blob/57bbc5e0d4b93483b8ae853352173516f1c08018/tensorflow/core/kernels/transpose_functor_cpu.cc#L103-L105
The check could be removed.

This PR fixes #46891.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow/python/kernel_tests/transpose_op_test.py [hunk 1]
```diff
@@ -525,6 +525,22 @@ def testError(self):
     self._testError(
         np.arange(0., 30).reshape([2, 3, 5]), [0, 1, 1], "2 is missing")
 
+  @test_util.run_v1_only("b/120545219")
```
Comment (rohan100jain) [inline tensorflow/python/kernel_tests/transpose_op_test.py:528]: I hope you don't need this decorator. Can you remove it and see if the tests pass?

File: tensorflow/python/kernel_tests/transpose_op_test.py [hunk 2]
```diff
@@ -525,6 +525,22 @@ def testError(self):
...
+  def testComplexScaler(self):
```
Comment (rohan100jain) [inline tensorflow/python/kernel_tests/transpose_op_test.py:529]: nit: s/Scaler/Scalar

File: tensorflow/python/kernel_tests/transpose_op_test.py [hunk 3]
```diff
@@ -525,6 +525,22 @@ def testError(self):
...
+    # Test case for GitHub issue 46891.
+    for dtype in [np.complex64, np.complex128]:
+      x = np.asarray(1+1j, dtype=dtype)
+      xt = self.evaluate(array_ops.transpose(x, conjugate=True))
+      self.assertAllEqual(xt, np.transpose(x).conj())
+
+  @test_util.run_v1_only("b/120545219")
```
Comment (rohan100jain) [inline tensorflow/python/kernel_tests/transpose_op_test.py:536]: ditto

**Review by rohan100jain — CHANGES_REQUESTED** [review]

File: tensorflow/python/kernel_tests/transpose_op_test.py (hunk 1 above)
Comment (yongtang) [reply tensorflow/python/kernel_tests/transpose_op_test.py]: Thanks @rohan100jain, Updated.

File: tensorflow/python/kernel_tests/transpose_op_test.py (hunk 2 above)
Comment (yongtang) [reply tensorflow/python/kernel_tests/transpose_op_test.py]: Updated.

File: tensorflow/python/kernel_tests/transpose_op_test.py (hunk 3 above)
Comment (yongtang) [reply tensorflow/python/kernel_tests/transpose_op_test.py]: Updated.

**Comment by yongtang** [comment]
@rohan100jain @cantonios thanks for the review. The PR has been updated. Please take a look.

**Review by cantonios — APPROVED** [review]
LGTM.

Ideally those extra tests would be folded into the original `testComplex64/128`, which already tests different tensor ranks.  Though with the current v1 decorator and the need to update these tests anyways for v2, this is probably fine for now.

**Review by cantonios — CHANGES_REQUESTED** [review]
@yongtang I updated the entire test to TF v2, so can you now merge your tests into `testComplex64()` and `testComplex128()`?

Something like this would work:
```
    self._testBoth(np.array(np.complex(1, 2)).astype(np.complex128))
    self._testBoth(np.complex(1, 2) * np.arange(0, 21).astype(np.complex128))
```

**Comment by yongtang** [comment]
Thanks @cantonios  for the help. The PR has been updated. Please take a look and let me know if there are any issues.

File: tensorflow/python/kernel_tests/transpose_op_test.py [hunk 4]
```diff
@@ -388,6 +388,8 @@ def testComplex64(self):
     self._testBoth(
         np.complex(1, 2) *
         np.arange(0, 1260).reshape([2, 3, 5, 7, 2, 3]).astype(np.complex64))
+    self._testBoth(np.array(np.complex(1, 2)).astype(np.complex64))
```
Comment (cantonios) [inline tensorflow/python/kernel_tests/transpose_op_test.py:391]: nit: put these above the others so they test in increasing rank order.

Comment (yongtang) [reply tensorflow/python/kernel_tests/transpose_op_test.py]: Thanks @cantonios. Done.

**Comment by yongtang** [comment]
Thanks @cantonios for the help. The PR has been updated. Please let me know if there are any additional issues.

**Review by cantonios — APPROVED** [review]

**Pull request merged**